lib/puppet/indirector/catalog/compiler.rb

```diff
@@ -213,26 +242,28 @@ def compile(node, options)
     config = nil
 
     benchmark(:notice, str) do
-      Puppet::Util::Profiler.profile(str, [:compiler, :compile, node.environment, node.name]) do
+      compile_type = checksum_type ? :static_compile : :compile
+      Puppet::Util::Profiler.profile(str, [:compiler, compile_type, node.environment, node.name]) do
```

Comment:
@joshcooper the value would be in distinguishing static compiles from non-static compiles